src/zabapgit_2fa.prog.abap

```diff
@@ -0,0 +1,659 @@
+  ENDMETHOD.
+
+  METHOD set_list_token_request.
+    ii_request->set_header_field( name  = if_http_header_fields_sap=>request_uri
+                                  value = gc_restendpoint_authorizations ).
+    ii_request->set_method( if_http_request=>co_request_method_get ).
+  ENDMETHOD.
+
+  METHOD set_del_token_request.
+    DATA: lv_url TYPE string.
+
+    lv_url = |{ gc_restendpoint_authorizations }/{ iv_token_id }|.
+
+    ii_request->set_header_field( name  = if_http_header_fields_sap=>request_uri
+                                  value = lv_url ).
+    " Other methods than POST and GET do not have constants unfortunately
+    " ii_request->set_method( if_http_request=>co_request_method_delete ).
+    ii_request->set_method( 'DELETE' ).
+  ENDMETHOD.
+
+  METHOD get_token_from_response.
+    CONSTANTS: lc_search_regex TYPE string VALUE `.*"token":"([^"]*).*$`.
+    DATA: lv_response TYPE string,
+          lo_regex    TYPE REF TO cl_abap_regex,
+          lo_matcher  TYPE REF TO cl_abap_matcher.
+
+    lv_response = cl_abap_codepage=>convert_from( ii_response->get_data( ) ).
+
+    CREATE OBJECT lo_regex
+      EXPORTING
+        pattern = lc_search_regex.
+
+    lo_matcher = lo_regex->create_matcher( text = lv_response ).
+    IF lo_matcher->match( ) = abap_true.
+      rv_token = lo_matcher->get_submatch( 1 ).
+    ENDIF.
+  ENDMETHOD.
+
+  METHOD get_tobedel_tokens_from_resp.
+    CONSTANTS: lc_search_regex TYPE string
+               VALUE `\{"id": ?(\d+)[^\{]*"app":\{[^\{^\}]*\}[^\{]*"fingerprint": ?` &
+               `"abapGit2FA"[^\{]*\}`.
+    DATA: lv_response TYPE string,
+          lo_regex    TYPE REF TO cl_abap_regex,
+          lo_matcher  TYPE REF TO cl_abap_matcher.
+
+    lv_response = cl_abap_codepage=>convert_from( ii_response->get_data( ) ).
+
+    CREATE OBJECT lo_regex
+      EXPORTING
+        pattern = lc_search_regex.
+
+    lo_matcher = lo_regex->create_matcher( text = lv_response ).
+    WHILE lo_matcher->find_next( ) = abap_true.
+      APPEND lo_matcher->get_submatch( 1 ) TO rt_ids.
+    ENDWHILE.
+  ENDMETHOD.
+
+  METHOD parse_repo_from_url.
+    CONSTANTS: lc_search_regex TYPE string VALUE 'https?:\/\/(www\.)?github.com\/(.*)$'.
+    DATA: lo_regex   TYPE REF TO cl_abap_regex,
+          lo_matcher TYPE REF TO cl_abap_matcher.
+
+    CREATE OBJECT lo_regex
+      EXPORTING
+        pattern = lc_search_regex.
+
+    lo_matcher = lo_regex->create_matcher( text = iv_url ).
+    IF lo_matcher->match( ) = abap_true.
+      rv_repo_name = lo_matcher->get_submatch( 1 ).
+    ELSE.
+      rv_repo_name = '???' ##NO_TEXT.
+    ENDIF.
+  ENDMETHOD.
+
+  METHOD get_service_id_from_url.
+    rv_id = 'github'.
+  ENDMETHOD.
+
+  METHOD is_2fa_required.
+    DATA: li_client       TYPE REF TO if_http_client,
+          lv_header_value TYPE string.
+
+    cl_http_client=>create_by_url(
+      EXPORTING
+        url    = gc_github_api_url
+      IMPORTING
```

Comment:
add following, I get a dump due to untrusted SSL certificates, as it currently does not use the same PSE(?) as the other abapGit requests
```
        ssl_id        = 'ANONYM'
        proxy_host    = lo_settings->get_proxy_url( )
        proxy_service = lo_settings->get_proxy_port( )
```

see example in include ZABAPGIT_HTTP